### PR TITLE
Fix: moderation app updates

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- removes block label from profile drop down
- updates profile card props in  moderation app
- decomposes useModeration hook

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/hooks/src/use-moderation.ts [hunk 1]
```diff
@@ -71,7 +74,63 @@ const createModerationMutation = async ({ dataToSign, contentId, contentType, ur
...
+          key.startsWith(PENDING_CACHE_KEY_PREFIX) ||
+          key.startsWith(MODERATED_CACHE_KEY_PREFIX) ||
+          key.startsWith(MODERATION_COUNT_CACHE_KEY_PREFIX)
+        ) {
+          uiCache.delete(key);
+        }
+      }
+
+      // refetch queries
+      queryClient.refetchQueries(PENDING_ITEMS_KEY);
```
Comment (kenshyx) [inline ui/hooks/src/use-moderation.ts]: these are async methods

File: ui/hooks/src/use-moderation.ts [hunk 2]
```diff
@@ -71,7 +74,63 @@ const createModerationMutation = async ({ dataToSign, contentId, contentType, ur
...
 function useModeration() {
   const queryClient = useQueryClient();
+  const sdk = getSDK();
   return useMutation((param: UseModerationParam) => createModerationMutation(param), {
+    onSuccess: () => {
+      // remove keys from cache
+      const uiCache = sdk.services.stash.getUiStash();
+      for (const key of uiCache.keys()) {
+        if (
+          key.startsWith(PENDING_CACHE_KEY_PREFIX) ||
```
Comment (kenshyx) [inline ui/hooks/src/use-moderation.ts]: you should guard for undefined or null keys

File: ui/hooks/src/use-moderation.ts [hunk 3]
```diff
@@ -71,7 +74,63 @@ const createModerationMutation = async ({ dataToSign, contentId, contentType, ur
...
+    const resp = await lastValueFrom(sdk.api.auth.signData(dataToSign));
+    const data = {
+      contentId,
+      contentType,
+      data: dataToSign,
+      signature: btoa(String.fromCharCode.apply(null, resp.data.signature)),
+    };
+
+    const status = await createModeration(url, data);
+    if (status === 409) {
```
Comment (kenshyx) [inline ui/hooks/src/use-moderation.ts]: consider using a switch statement here

File: ui/hooks/src/use-moderation.ts [hunk 4]
```diff
@@ -56,15 +56,18 @@ const createModerationMutation = async ({ dataToSign, contentId, contentType, ur
...
-      throw new Error('Unable to process your request right now. Please try again later');
+
+    switch (true) {
+      case status === 409:
+        throw new Error(`This content has already been moderated by you`);
+      case status === 403:
+        throw new Error('You are not authorized to perform this operation');
+      case status === 400:
+        throw new Error('Bad request. Please try again later');
+      case status >= 400:
```
Comment (kenshyx) [inline ui/hooks/src/use-moderation.ts:67]: there is no `default` case?

Comment (josenriagu) [reply ui/hooks/src/use-moderation.ts:67]: no, I have handled all possible status values from the endpoint

Comment (kenshyx) [reply ui/hooks/src/use-moderation.ts:67]: a switch statement should always have a default return..., in this case add something that will flag that will help with cases that are not caught by the exact value comparison

**Review by kenshyx — APPROVED** [review]

**Pull request merged**